tensorflow/compiler/xla/service/cpu/simple_orc_jit.cc

```diff
@@ -41,6 +41,11 @@ limitations under the License.
 #include "tensorflow/compiler/xla/types.h"
 #include "tensorflow/core/platform/logging.h"
 
+#if defined(__APPLE__)
+static void sincos(double, double*, double*)  __attribute__((weakref ("__sincos")));
```

Comment:
I'm wary of using non-standard attributes here that aren't used elsewhere in TensorFlow.

Can we instead  handle this by `REGSTER_CPU_RUNTIME_SYMBOL`ing differently on `__APPLE__`?  I don't think we need the name anywhere in LLVM IR generation (so no need for a `GetSinCosFunctionName()` helper) and if the LLVM optimizer generates a call to sincos(f) it should use the correct symbol name for the compilation target.

Reply:
We could do that. So should I define the runtime symbol in cpu_runtime.cc or should we make a new cpu_runtime_apple.cc (or something like that based on _neon _avx etc)

Reply:
Special-casing this in `RegisterKnownJITSymbols` seems fine for now.